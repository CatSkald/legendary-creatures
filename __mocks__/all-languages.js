import languages from "../src/i18n/languages";

exports.allLanguages = Object.entries(languages).map(
  ([lang, langProps]) => langProps,
);
