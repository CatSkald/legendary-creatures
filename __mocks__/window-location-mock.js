exports.mockWindowLocation = () => {
  const originalWindowLocation = window.location;

  beforeEach(() => {
    delete window.location;

    // mock window.location
    window.location = Object.defineProperties(
      {},
      {
        ...Object.getOwnPropertyDescriptors(originalWindowLocation),
        assign: {
          configurable: true,
          value: jest.fn(),
        },
      },
    );
  });

  afterEach(() => {
    // restore original `jsdom` location
    window.location = originalWindowLocation;
  });
};
