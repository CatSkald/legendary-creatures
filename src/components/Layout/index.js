import "../../styles/global.scss";
import "../../styles/theme-light.scss";
import "../../styles/theme-dark.scss";
import styles from "./index.module.scss";

import React, { useState } from "react";
import Header from "../Header";
import Footer from "../Footer";

const LocaleContext = React.createContext();

const Layout = ({ children, pageContext: { language } }) => {
  const [isDarkColorTheme, switchColorTheme] = useState(false);

  function handleToggleColorTheme() {
    switchColorTheme(!isDarkColorTheme);
  }

  return (
    <LocaleContext.Provider value={{ language }}>
      <div
        className={`body ${styles.Wrapper} ${
          isDarkColorTheme ? "theme-dark" : "theme-light"
        }`}
      >
        <Header
          isDarkColorTheme={isDarkColorTheme}
          handleToggleColorTheme={handleToggleColorTheme}
        />
        <section className={styles.SiteContent} role="main">
          <div className={styles.Container}>{children}</div>
        </section>
        <Footer />
      </div>
    </LocaleContext.Provider>
  );
};

export { Layout, LocaleContext };
