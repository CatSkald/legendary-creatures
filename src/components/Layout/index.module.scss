.layout__container {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.layout {
  display: flex;
  flex: 1;
  width: 100%;
}

.layout__content {
  width: 100%;
  max-width: var(--container-width);
  margin: 0 auto;
  padding: var(--space-sm);

  @media screen and (min-width: 768px /*--screen-medium*/) {
    padding: var(--space);
  }
}
