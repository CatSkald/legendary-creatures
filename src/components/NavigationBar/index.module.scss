.active {
  /* placeholder to be referenced from styles */
}

.navigationBar {
  display: none;
  flex-direction: column;
  padding: 0 var(--space-sm);

  &.active {
    display: flex;
  }

  @media screen and (min-width: 768px /*--screen-medium*/) {
    display: flex;
    flex-direction: row;
    align-items: center;
  }
}

.navigationBar__link {
  color: var(--menu-text);
  text-decoration: none;
  position: relative;
  padding: 0;
  margin-top: var(--space-xxs);
  text-align: center;

  @media screen and (min-width: 768px /*--screen-medium*/) {
    margin: var(--space-sm);

    &:hover,
    &.active,
    &.active:hover {
      &:after {
        opacity: 1;
        width: 100%;
      }
    }

    &:after {
      content: "";
      display: inline-block;
      width: 0;
      height: 0.4rem;
      background: var(--menu-text);
      bottom: calc(0px - var(--space-sm));
      position: absolute;
      left: 0;
      opacity: 0;
      transition: 0.3s ease-in-out;
    }
  }

  &.active {
    font-weight: bold;
    color: var(--menu-text-selected);

    @media screen and (min-width: 768px /*--screen-medium*/) {
      color: var(--menu-text);
    }
  }

  &:hover,
  &.active:hover {
    color: var(--menu-text-hover);
  }
}
