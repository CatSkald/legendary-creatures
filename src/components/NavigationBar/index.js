import {
  navigationBar,
  active,
  navigationBar__link,
} from "./index.module.scss";

import React from "react";
import PropTypes from "prop-types";

import LocalizedLink from "../LocalizedLink";
import { LocaleContext } from "../Layout";
import { pages } from "../../i18n/navigation";

const NavigationBar = (props) => {
  const { language } = React.useContext(LocaleContext);
  var menuItems = Object.entries(pages)
    .filter(([page, data]) => data.type === "menu")
    .map(([page, data]) => data[language.code]);

  return (
    <>
      <nav className={`${navigationBar} ${props.isActive ? active : ""}`}>
        {menuItems.map((menu) => (
          <LocalizedLink
            className={navigationBar__link}
            key={menu.path}
            to={menu.path}
            aria-label={menu.title}
            activeClassName={active}
            onClick={() => props.handleToggleMenu()}
          >
            {menu.title}
          </LocalizedLink>
        ))}
      </nav>
    </>
  );
};

NavigationBar.propTypes = {
  handleToggleMenu: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
};

export default NavigationBar;
