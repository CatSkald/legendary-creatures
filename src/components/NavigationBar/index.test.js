import React from "react";
import { render } from "@testing-library/react";

import { allLanguages } from "../../../__mocks__/all-languages";
import { LocaleContext } from "../Layout";
import NavigationBar from "../NavigationBar";

describe("NavigationBar", () => {
  const handleToggleMenuMock = jest.fn();

  beforeEach(() => {
    handleToggleMenuMock.mockClear();
  });

  describe.each(allLanguages)(
    "renders correctly for language %s",
    (language) => {
      test.each([true, false])("when isActive %s", (isActive) => {
        const { container } = render(
          <LocaleContext.Provider value={{ language }}>
            <NavigationBar
              handleToggleMenu={handleToggleMenuMock}
              isActive={isActive}
            />
          </LocaleContext.Provider>,
        );
        expect(container).toMatchSnapshot();
      });
    },
  );
});
