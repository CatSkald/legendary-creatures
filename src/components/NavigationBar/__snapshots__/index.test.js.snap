// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders correctly for language {
  name: 'Українська',
  code: 'uk',
  path: '/uk',
  dateFormat: 'DD.MM.YYYY'
} when isActive false 1`] = `
<div>
  <nav
    class="navigationBar "
  >
    <a
      aria-label="Головна"
      class="navigationBar__link"
      href="/uk"
      hreflang="uk"
    >
      Головна
    </a>
    <a
      aria-label="Бібліотека істот"
      class="navigationBar__link"
      href="/uk/істоти"
      hreflang="uk"
    >
      Бібліотека істот
    </a>
    <a
      aria-label="Мітки"
      class="navigationBar__link"
      href="/uk/мітки"
      hreflang="uk"
    >
      Мітки
    </a>
    <a
      aria-label="Про проект"
      class="navigationBar__link"
      href="/uk/about"
      hreflang="uk"
    >
      Про проект
    </a>
  </nav>
</div>
`;

exports[`NavigationBar renders correctly for language {
  name: 'Українська',
  code: 'uk',
  path: '/uk',
  dateFormat: 'DD.MM.YYYY'
} when isActive true 1`] = `
<div>
  <nav
    class="navigationBar active"
  >
    <a
      aria-label="Головна"
      class="navigationBar__link"
      href="/uk"
      hreflang="uk"
    >
      Головна
    </a>
    <a
      aria-label="Бібліотека істот"
      class="navigationBar__link"
      href="/uk/істоти"
      hreflang="uk"
    >
      Бібліотека істот
    </a>
    <a
      aria-label="Мітки"
      class="navigationBar__link"
      href="/uk/мітки"
      hreflang="uk"
    >
      Мітки
    </a>
    <a
      aria-label="Про проект"
      class="navigationBar__link"
      href="/uk/about"
      hreflang="uk"
    >
      Про проект
    </a>
  </nav>
</div>
`;

exports[`NavigationBar renders correctly for language { name: 'English', code: 'en', path: '', dateFormat: 'DD/MM/YYYY' } when isActive false 1`] = `
<div>
  <nav
    class="navigationBar "
  >
    <a
      aria-label="Home"
      class="navigationBar__link"
      href="/"
      hreflang="en"
    >
      Home
    </a>
    <a
      aria-label="Creatures"
      class="navigationBar__link"
      href="/creatures"
      hreflang="en"
    >
      Creatures
    </a>
    <a
      aria-label="Tags"
      class="navigationBar__link"
      href="/tags"
      hreflang="en"
    >
      Tags
    </a>
    <a
      aria-label="About"
      class="navigationBar__link"
      href="/about"
      hreflang="en"
    >
      About
    </a>
  </nav>
</div>
`;

exports[`NavigationBar renders correctly for language { name: 'English', code: 'en', path: '', dateFormat: 'DD/MM/YYYY' } when isActive true 1`] = `
<div>
  <nav
    class="navigationBar active"
  >
    <a
      aria-label="Home"
      class="navigationBar__link"
      href="/"
      hreflang="en"
    >
      Home
    </a>
    <a
      aria-label="Creatures"
      class="navigationBar__link"
      href="/creatures"
      hreflang="en"
    >
      Creatures
    </a>
    <a
      aria-label="Tags"
      class="navigationBar__link"
      href="/tags"
      hreflang="en"
    >
      Tags
    </a>
    <a
      aria-label="About"
      class="navigationBar__link"
      href="/about"
      hreflang="en"
    >
      About
    </a>
  </nav>
</div>
`;

exports[`NavigationBar renders correctly for language { name: 'Русский', code: 'ru', path: '/ru', dateFormat: 'DD.MM.YYYY' } when isActive false 1`] = `
<div>
  <nav
    class="navigationBar "
  >
    <a
      aria-label="Главная"
      class="navigationBar__link"
      href="/ru"
      hreflang="ru"
    >
      Главная
    </a>
    <a
      aria-label="Библиотека существ"
      class="navigationBar__link"
      href="/ru/существа"
      hreflang="ru"
    >
      Библиотека существ
    </a>
    <a
      aria-label="Метки"
      class="navigationBar__link"
      href="/ru/метки"
      hreflang="ru"
    >
      Метки
    </a>
    <a
      aria-label="О проекте"
      class="navigationBar__link"
      href="/ru/about"
      hreflang="ru"
    >
      О проекте
    </a>
  </nav>
</div>
`;

exports[`NavigationBar renders correctly for language { name: 'Русский', code: 'ru', path: '/ru', dateFormat: 'DD.MM.YYYY' } when isActive true 1`] = `
<div>
  <nav
    class="navigationBar active"
  >
    <a
      aria-label="Главная"
      class="navigationBar__link"
      href="/ru"
      hreflang="ru"
    >
      Главная
    </a>
    <a
      aria-label="Библиотека существ"
      class="navigationBar__link"
      href="/ru/существа"
      hreflang="ru"
    >
      Библиотека существ
    </a>
    <a
      aria-label="Метки"
      class="navigationBar__link"
      href="/ru/метки"
      hreflang="ru"
    >
      Метки
    </a>
    <a
      aria-label="О проекте"
      class="navigationBar__link"
      href="/ru/about"
      hreflang="ru"
    >
      О проекте
    </a>
  </nav>
</div>
`;
